templates/adherent/set_email_notifications.html.twig

```diff
@@ -16,10 +16,75 @@
+            var selectDistanceNotificationPc = $('#adherent_email_subscription_citizenProjectCreationEmailSubscriptionRadius'),
+                checkBoxNotificationPc = $('#adherent_email_subscription_emails_subscriptions_3'),
+                labelCheckBoxNotificationPc = $('label[for="adherent_email_subscription_emails_subscriptions_3"]');
+
+            checkboxHandler();
+
+            function checkboxHandler() {
+                const isCheck = checkBoxNotificationPc.is(':checked');
+
+                if (isCheck) {
+                    labelCheckBoxNotificationPc.text('');
+                    labelCheckBoxNotificationPc.append(
+                        $('<span>')
+                            .append('Être notifié(e) de la création de nouveaux projets citoyens dans un rayon de ')
+                            .append(selectDistanceNotificationPc)
+                            .append('autour de chez moi')
+                    );
+                    selectDistanceNotificationPc.children('option:selected').removeAttr("selected");
+                    selectDistanceNotificationPc.children('option[value="{{ app.user.citizenProjectCreationEmailSubscriptionRadius != constant('DISABLED_CITIZEN_PROJECT_EMAIL', app.user) ? app.user.citizenProjectCreationEmailSubscription : 10}}"]').prop('selected', true);
+                    selectDistanceNotificationPc.show();
+                } else {
+                    selectDistanceNotificationPc.hide();
+                    $("form[name='adherent_email_subscription']").append(selectDistanceNotificationPc);
+                    labelCheckBoxNotificationPc.text('Être notifié(e) de la création de nouveaux projets citoyens');
+                }
+            }
+
+            function selectHandler() {
+                var val = selectDistanceNotificationPc.find('option:selected').val();
+                console.log(val);
+                if (val == '0') {
+                    labelCheckBoxNotificationPc.text('');
+                    labelCheckBoxNotificationPc.append(
+                        $('<span>')
+                            .append('Être notifié(e) de ')
+                            .append(selectDistanceNotificationPc)
+                            .append('création de projet citoyen autour de chez moi')
+                    );
+                    selectDistanceNotificationPc.unbind('change').on('change', selectHandler);
+                } else {
+                    labelCheckBoxNotificationPc.text('');
+                    labelCheckBoxNotificationPc.append(
+                        $('<span>')
+                            .append('Être notifié(e) de la création de nouveaux projets citoyens dans un rayon de ')
+                            .append(selectDistanceNotificationPc)
+                            .append('autour de chez moi')
+                    );
+                    selectDistanceNotificationPc.unbind('change').on('change', selectHandler);
+                }
```

Comment:
you could avoid code duplication here by only constructing the message within the condition.

```
if (val == '0') {
    var message = 'Être notifié(e) de toute création de projet citoyen autour de chez moi';
} else {
    var message = 'Être notifié(e) de la création de nouveaux projets citoyens dans un rayon de '+selectDistanceNotificationPc+' autour de chez moi';
}

// apply text and reapply selectHandler